### PR TITLE
spidermonkey: depend on python@3.10

### PR DESCRIPTION
**Pull request by branchv**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

---

For #93940

### CODE REVIEW

File: Formula/spidermonkey.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ class Spidermonkey < Formula
   version "1.8.5"
   sha256 "5d12f7e1f5b4a99436685d97b9b7b75f094d33580227aa998c406bbae6f2a687"
   license "MPL-1.1"
-  revision 4
+  revision 5
```
[the remark this answers is not in the document]
Comment (branchvincent) [reply Formula/spidermonkey.rb:8]: good point, will drop

**Comment by branchvincent** [comment]
`.../js/src/methodjit/MethodJIT.cpp:192:5: error: meaningless 'volatile' on asm outside function`

**Comment by cho-m** [comment]
I wonder if this version is too old now since it has been around for over 10+ years.

Probably need to retry updating like in #78997.
The problem is going to be `couchdb` which will require a versioned copy of `spidermonkey`. May consider using a resource there since the spidermonkey version should be updated along with `couchdb` whenever possible.

**Comment by cho-m** [comment]
So `spidermonkey` should now be updated to newer version v91 with Python 3 in #94905.

One thing that needs to be revisited is that I used `python@3.9` rather than `python@3.10` due to error:
```
==> ../configure --prefix=/usr/local/Cellar/spidermonkey@91/91.6.0 --enable-optimize --enable-readline ...
Last 15 lines from /Users/cho-m/Library/Logs/Homebrew/spidermonkey@91/02.configure:
  File "/private/tmp/spidermonkeyA91-20220210-6205-qg8q9a/firefox-91.6.0/python/mozbuild/mozbuild/util.py", line 1050, in method_call
    cache[args] = self.func(instance, *args)

   ...

  File "/private/tmp/spidermonkeyA91-20220210-6205-qg8q9a/firefox-91.6.0/python/mozbuild/mozbuild/virtualenv.py", line 615, in _disable_pip_outdated_warning
    with open(os.path.join(site_packages, pip_dist_info, "INSTALLER"), "w") as file:
PermissionError: [Errno 1] Operation not permitted: '/usr/local/lib/python3.10/site-packages/pip-22.0.2.dist-info/INSTALLER'
```

This is probably a side-effect of how we set up `python@3.10` since Linux distros are able to use latest Python

**Comment by cho-m** [comment]
Closing for now. Can reopen if you want to work on `python@3.9 --> python@3.10` update.

**Pull request closed without merging**